control-plane/connect-inject/webhook/mesh_webhook.go

```diff
@@ -209,6 +215,12 @@ type multiPortInfo struct {
 	serviceName  string
 }
 
+func isWindows(pod corev1.Pod) bool {
+	podOS := pod.Spec.NodeSelector["kubernetes.io/os"]
```

Comment:
Because it's an optional value, we should check if it's set:
```suggestion
       podOS, ok := pod.Spec.NodeSelector["kubernetes.io/os"]
	if !ok {
		return false
	}
```